clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -100,6 +106,16 @@ protected boolean removeEldestEntry(Map.Entry<Schema, GenericRecord> eldest) {
 
   private static final ThreadLocal<ReusableObjects> REUSABLE_OBJECTS = ThreadLocal.withInitial(ReusableObjects::new);
 
+  /**
+   * The following two fields are used to speed up the requests with a big number of keys:
+   * 1. Split the big request into smaller chunks.
+   * 2. Execute these chunks concurrently.
+   */
+  private static final ExecutorService READ_CHUNK_EXECUTOR = Executors.newFixedThreadPool(
+      Runtime.getRuntime().availableProcessors(),
```

Comment:
I don't have other good default number in mind, but I am wondering if we take up all threads, will it affect the follow up execution such as Frame transformation and other app computation?

Reply:
Synced offline, it should be good for now. It was validated and we will validate it in the new release as well.